### PR TITLE
Weather app

### PR DESCRIPTION
**Pull request by Mupa1**
## Weather App

In this project, I created a weather app using Javascript and HTML/CSS. I also fetched the weather information from OpenWeatherMap API. It contains:-
- An input field for searching by city name.
- A section that displays the temperature in Celcius by default.
- A section that displays the current date.
- A section that displays the weather description.
- Buttons for toggling the temperature between Celcius and Fahrenheit.

[Live Link](https://raw.githack.com/Mupa1/weather-app/weather_app/dist/index.html)

Kindly review my project.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  domElements.showDescription(weather);
+  displayDate(weather);
+  switchTemps(weather);
+};
+
+const fetchResults = (city) => {
+  fetch(`${baseUrl}weather?q=${city}&units=metric&APPID=${key}`)
+    .then(weather => weather.json())
+    .then(displayResults);
+};
```
Comment (uche-inyama) [inline src/index.js]: Add error handling for your promises.

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi Mupa :wave: 

Your project works as it should. 

However, there are few additions to make.

- [x] Add error handling for `promises` or `async/await` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))

- [x] Install webpack-dev-server with npm or yarn, then add this to your script property in package.json:

``` 
"scripts": {
  
       "start": "webpack-dev-server --mode development --open" 
  },
```

#### After implementing the changes submit another form for a code review.

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Inyama Henry ```

File: src/index.js (hunk 1 above)
Comment (Mupa1) [reply src/index.js]: Thank you @uche-inyama for your feedback. I have made all the changes you requested and requested for another code review.

**Review by MarioDena — APPROVED** [review]
Hello! Great work implementing the requested changes!

Project approved! 🟢

**Pull request merged**